clang/lib/AST/ExprConstant.cpp

```diff
@@ -9749,6 +9749,11 @@ bool LValueExprEvaluator::VisitArraySubscriptExpr(const ArraySubscriptExpr *E) {
 
     if (Success) {
       Result.setFrom(Info.Ctx, Val);
+      // If Index cannot be represented as a 64 bit integer, return
+      // unsuccessful.
+      if (!Index.tryExtValue().has_value())
```

Comment:
So, with this change we exit constexpr evaluation with some kind of error. I see that `Error` function is actually intended to emit notes additionally to a some error which according to the test we don't emit, so if we don't want to emit messages for the case I don't think using `Error` is right here.

Does CodeGen produce a correct result for the code? In case it is not, the fix is actually broader, so I'd also suggest a test checking LLVM IR emitted.

Reply:
See https://github.com/llvm/llvm-project/issues/63562